packages/access-client/src/agent-use-cases.js

```diff
@@ -115,38 +115,37 @@ export async function pollAccessClaimUntil(
   opts
 ) {
   const interval = opts?.interval || 250
-  /** @type {ReturnType<typeof setTimeout>} */
-  let pollingTimeoutId
   const claimed = await new Promise((resolve, reject) => {
     opts?.signal?.addEventListener('abort', (e) => {
-      pollingTimeoutId && clearTimeout(pollingTimeoutId)
-      reject(
-        new Error('pollAccessClaimUntilSessionProof aborted', { cause: e })
-      )
+      reject(new Error('pollAccessClaimUntil aborted', { cause: e }))
     })
     poll(interval)
     /**
      * @param {number} retryAfter
      */
     async function poll(retryAfter) {
-      const pollClaimResult = await access.invokeAndExecute(
-        w3caps.Access.claim,
-        { with: delegee }
-      )
-      if (pollClaimResult.error) {
-        return reject(pollClaimResult)
-      }
-      try {
-        const claimedDelegations = Object.values(
-          pollClaimResult.delegations
-        ).flatMap((d) => bytesToDelegations(d))
-        if (delegationsMatch(claimedDelegations)) {
-          resolve(claimedDelegations)
-        } else {
-          pollingTimeoutId = setTimeout(() => poll(retryAfter), retryAfter)
+      if (opts?.signal?.aborted) {
+        reject(new Error('pollAccessClaimUntil aborted'))
+      } else {
+        const pollClaimResult = await access.invokeAndExecute(
+          w3caps.Access.claim,
+          { with: delegee }
+        )
+        if (pollClaimResult.error) {
+          return reject(pollClaimResult)
+        }
+        try {
+          const claimedDelegations = Object.values(
+            pollClaimResult.delegations
+          ).flatMap((d) => bytesToDelegations(d))
+          if (delegationsMatch(claimedDelegations)) {
+            resolve(claimedDelegations)
+          } else {
+            setTimeout(() => poll(retryAfter), retryAfter)
+          }
+        } catch (error) {
+          reject(error)
```

Comment:
I don't understand why so complicated? Specifically, the delayed recursive function call and mixing `await` with `resolve`/`reject`. Why not just a `while` loop?

```js
while (true) {
  if (opts?.signal?.aborted) throw opts.signal.reason ?? new Error('operation aborted')
  const res = await access.invokeAndExecute(w3caps.Access.claim, { with: delegee })
  if (res.error) throw res
  const claims = Object.values(res.delegations).flatMap(d => bytesToDelegations(d))
  if (delegationsMatch(claims)) return claims
  await new Promise(resolve => setTimeout(resolve, retryAfter))
}
```

Reply:
Not blocking btw, looks like what you have here will fix the issue just fine 😇

Reply:
hm no I agree tho, I'll take a crack at reworking this a bit - I didn't really touch the structure of this code during this fix but it is a bit odd and I'm not sure it's needed - will take your snippet for a spin!

Reply:
ok yea this worked great, thanks!!